### PR TITLE
Add `renameFile` and `renameFileSync`, `cwd` option 

### LINKED ISSUE #22: Add a convenience method for renaming a file in a directory
Given a path to a directory, it's a little cumbersome to rename a single file:

```js
import {moveFile} from 'move-file';
await moveFile(`${directoryPath}/unicorn.png`, `${directoryPath}/unicorns.png`);

// or

import fs from 'node:fs/promises';
await fs.rename(`${directoryPath}/unicorn.png`, `${directoryPath}/unicorns.png`);
```

This only gets noisier if using `path.resolve`, and the actual change can get lost or accidentally replaced.

Consider (with a `cwd` option like `cpy` or `del`):

```js
import {renameFile} from 'move-file';
await renameFile('unicorn.png', 'unicorns.png', {cwd: directoryPath});
```

### PR DESCRIPTION
**Pull request by tommy-mitchell**
Resolves #22.

Adds a `cwd` option (like `cpy` or `del`) and adds wrappers around `moveFile` and `moveFileSync` that makes it nicer to rename a file:

```js
import {renameFile} from 'move-file';

await renameFile('unicorn.png', 'unicorns.png', {cwd: 'source'});
console.log('The file has been renamed');
```

I haven't added the new functions to the readme because I'm not sure how they should be formatted with the options being common to all of them.

I could also update the deps, if you'd like.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -53,3 +72,11 @@ export function moveFileSync(sourcePath, destinationPath, {overwrite = true, dir
 		}
 	}
 }
+
+export async function renameFile(source, destination, options) {
+	return moveFile(source, destination, options);
```
Comment (sindresorhus) [inline index.js:77]: Should it validate that the source and destination are in the same directory?

**Comment by sindresorhus** [comment]
> I haven't added the new functions to the readme because I'm not sure how they should be formatted with the options being common to all of them.

Maybe make the options a `###` section called `Options`.

File: index.js (hunk 1 above)
Comment (tommy-mitchell) [reply index.js:77]: I guess it depends on if `renameFile` is an alias to `moveFile` or if it only renames a file within a directory. Based on the name, the latter probably makes more sense.

Comment (sindresorhus) [reply index.js:77]: Yeah, lets validate it.

**Comment by tommy-mitchell** [comment]
Went ahead and bumped dependencies as well. `temp-write` can probably be dropped for `tempy`, but I only did so for the new rename tests.

File: index.d.ts [hunk 2]
```diff
@@ -48,3 +57,36 @@ console.log('The file has been moved');
...
+
+@example
+```
+import {renameFile} from 'move-file';
+
+await renameFile('unicorn.png', 'unicorns.png', {cwd: 'source'});
+console.log('The file has been renamed');
+```
+*/
+export function renameFile(source: string, destination: string, options?: Options): Promise<void>;
```
Comment (sindresorhus) [inline index.d.ts:76]:
```suggestion
export function renameFile(sourceFilename: string, destinationFilename: string, options?: Options): Promise<void>;
```

Just to make it clear it has to be a filename, not a path?

Comment (tommy-mitchell) [reply index.d.ts:76]: I think letting it be a path lets users choose the more ergonomic version for themselves:

```js
await renameFile('../unicorn.txt', '../unicorns.txt');
```

```js
await renameFile('unicorn.txt', 'unicorns.txt', {cwd: path.resolve('some/long/nested/path')});
```

Comment (sindresorhus) [reply index.d.ts:76]: 👍

**Pull request merged**